coconut/control/control.go

```diff
@@ -262,56 +265,62 @@ func GetEnvironments(cxt context.Context, rpc *coconut.RpcClient, cmd *cobra.Com
 func CreateEnvironment(cxt context.Context, rpc *coconut.RpcClient, cmd *cobra.Command, args []string, o io.Writer) (err error) {
 	configPayload, err := cmd.Flags().GetString("configuration")
 	if err != nil {
-		err = errors.New("cannot get configuration payload value")
-		return
+		return fmt.Errorf("cannot get configuration payload value: %w", err)
 	}
 	userWfPath, err := cmd.Flags().GetString("workflow-template")
 	if err != nil {
-		err = errors.New("cannot get workflow template value")
-		return
+		return fmt.Errorf("cannot get workflow template value: %w", err)
 	}
 	if cmd.Flags().Changed("configuration") && len(configPayload) == 0 && cmd.Flags().Changed("workflow-template") && len(userWfPath) == 0 {
-		err = errors.New("no configuration payload or workflow template provided")
-		return
+		return fmt.Errorf("no configuration payload or workflow template provided: %w", err)
 	}
 
 	payloadData := new(ConfigurationPayload)
 	var wfPath string
 	if cmd.Flags().Changed("configuration") && len(configPayload) > 0 {
-		configPayloadDoc, err := os.ReadFile(configPayload)
-		if err != nil {
-			err = errors.New("cannot read configuration payload file")
-			return err
-		}
-		err = json.Unmarshal(configPayloadDoc, &payloadData)
-		if err != nil {
-			err = errors.New("cannot unmarshal configuration payload")
-			return err
+		if strings.HasSuffix(strings.ToLower(configPayload), ".json") {
+			configPayloadDoc, err := os.ReadFile(configPayload)
+			if err != nil {
+				return fmt.Errorf("cannot read local configuration payload: %w", err)
+			}
+			err = json.Unmarshal(configPayloadDoc, &payloadData)
+			if err != nil {
+				return fmt.Errorf("cannot unmarshal local configuration payload: %w", err)
+			}
+		} else {
+			if strings.HasPrefix(configPayload, HLCONFIG_PATH_PREFIX+HLCONFIG_COMPONENT_PREFIX) {
+				configPayload = strings.TrimPrefix(configPayload, HLCONFIG_PATH_PREFIX+HLCONFIG_COMPONENT_PREFIX)
+			}
+			configPayloadDoc, err := apricot.Instance().GetRuntimeEntry(HLCONFIG_COMPONENT_PREFIX, configPayload)
+			if err != nil {
+				return fmt.Errorf("cannot retrieve file from "+HLCONFIG_PATH_PREFIX+HLCONFIG_COMPONENT_PREFIX+": %w", err)
```

Comment:
I'd also include the actual key we're failing to retrieve (i.e. the contents of `configPayload` that should really be called `configPayloadKey` or somesuch). Also, if it's on disk we call it a "file", if it's from Consul we call it a "payload", "entry" or "document".

Generally speaking about data resources:
* if it's on disk it's a file and it has a name;
* if it's in memory (in Consul, in a buffer, object storage, etc.) then it's a payload or document, and it has a key or path.